public/app/features/explore/RichHistory/RichHistoryCard.test.tsx

```diff
@@ -56,93 +79,140 @@ const starredQueryWithComment: RichHistoryQuery<MockQuery> = {
   ],
 };
 
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
 describe('RichHistoryCard', () => {
-  it('should render all queries', () => {
-    const wrapper = setup();
-    expect(wrapper.find({ 'aria-label': 'Query text' })).toHaveLength(3);
-    expect(wrapper.find({ 'aria-label': 'Query text' }).at(0).text()).toEqual('{"query":"query1"}');
-    expect(wrapper.find({ 'aria-label': 'Query text' }).at(1).text()).toEqual('{"query":"query2"}');
-    expect(wrapper.find({ 'aria-label': 'Query text' }).at(2).text()).toEqual('{"query":"query3"}');
-  });
-  it('should render data source icon', () => {
-    const wrapper = setup();
-    expect(wrapper.find({ 'aria-label': 'Data source icon' })).toHaveLength(1);
+  it('should render all queries', async () => {
+    setup();
+    const queries = await screen.findAllByLabelText('Query text');
+    expect(queries).toHaveLength(3);
+    expect(queries[0]).toHaveTextContent('query1');
+    expect(queries[1]).toHaveTextContent('query2');
+    expect(queries[2]).toHaveTextContent('query3');
   });
-  it('should render data source name', () => {
-    const wrapper = setup();
-    expect(wrapper.find({ 'aria-label': 'Data source name' }).text()).toEqual('Test datasource');
+  it('should render data source icon and name', async () => {
+    setup();
+    const datasourceIcon = await screen.findAllByLabelText('Data source icon');
+    const datasourceName = await screen.findAllByLabelText('Data source name');
+    expect(datasourceIcon).toHaveLength(1);
+    expect(datasourceName).toHaveLength(1);
```

Comment:
```suggestion
    const datasourceIcon = await screen.findByLabelText('Data source icon');
    const datasourceName = screen.getByLabelText('Data source name');
    expect(datasourceIcon).toBeInTheDocument();
    expect(datasourceName).toBeInTheDocument();
```
There's usually no need to use second `await` call, by the time that line is reached all the promises are resolved by the first one. Also since we're expecting one element to be rendered, `toBeInTheDocument` is a bit more explicit.

Reply:
Unfortunately in this case, it seems I need the second await. The object returns empty without it. I have made the other changes.

Reply:
Hmm, that's weird, I've run the test with this change locally and it works fine.

Reply:
🤷
https://user-images.githubusercontent.com/1533128/161766978-fed789a1-228f-4721-8341-a735f45ff86a.mov

Reply:
I see in console that you're using `findByLabelText` but it'd be `getByLabelText`.

Reply:
Yep, youre right! 🤦‍♀️ Brb, getting the [summary table](https://testing-library.com/docs/queries/about) tattooed on the inside of my eyelids